src/mats_l1_processing/read_parquet_functions.py

```diff
@@ -0,0 +1,255 @@
+    path: str,
+    filesystem: Optional[pa.fs.FileSystem] = None,
+) -> DataFrame:
+    """Reads the CCD data and metadata from the specified path or S3 bucket
+    between the specified times.
+
+    Args:
+        from (datetime):            Read CCD data from this timestamp (inclusive).
+        to (datetime):              Read CCD data up to this timestamp (inclusive).
+        path (str):                 Path to dataset. May be a directory or a
+                                    bucket, depending on filesystem.
+        filesystem (filesystem):    Optional. Filesystem to read. If not
+                                    specified will assume that path points to
+                                    an ordinary directory disk. (Default: None.)
+
+    Returns:
+        DataFrame:  The CCD data.
+    """
+
+    return ds.dataset(
+        path,
+        filesystem=filesystem,
+    ).to_table(filter=(
+        (ds.field("EXPDate") >= Timestamp(start))
+        & (ds.field("EXPDate") <= Timestamp(stop))
+    )).to_pandas().reset_index()
+
+
+def read_ccd_items_in_interval(
+    start: datetime,
+    stop: datetime,
+    path: str,
+    filesystem: Optional[pa.fs.FileSystem] = None,
+) -> List[CCDItem]:
+    """Reads the CCD data and metadata from the specified path or S3 bucket
+    between the specified times.
+    Some column names are translated and some convenience data is added before
+    data is converted and and returned.
+
+    Args:
+        from (datetime):            Read CCD data from this timestamp (inclusive).
+        to (datetime):              Read CCD data up to this timestamp (inclusive).
+        path (str):                 Path to dataset. May be a directory or a
+                                    bucket, depending on filesystem.
+        filesystem (filesystem):    Optional. Filesystem to read. If not
+                                    specified will assume that path points to
+                                    an ordinary directory disk. (Default: None.)
+
+    Returns:
+        list[dict[str, Any]]:   List of dictionary representations of the CCD data.
+    """
+
+    return dataframe_to_ccd_items(
+        read_ccd_data_in_interval(start, stop, path, filesystem)
+    )
+
+
+def read_ccd_data(
+    path: str,
+    filesystem: Optional[pa.fs.FileSystem] = None,
+) -> DataFrame:
+    """Reads the CCD data and metadata from a singel file at the specified path.
+
+    Args:
+        path (str):                 Path to dataset. May be a directory or a
+                                    bucket, depending on filesystem.
+        filesystem (filesystem):    Optional. Filesystem to read. If not
+                                    specified will assume that path points to
+                                    an ordinary directory disk. (Default: None.)
+
+    Returns:
+        DataFrame:  The CCD data.
+    """
+
+    return pq.read_table(
+        path,
+        filesystem=filesystem,
+    ).to_pandas().reset_index()
```

Comment:
Since we are living in a hypothetical world it's uncertain right now what is the wanted behaviour, but I'm just noting that this will return a dataframe with integer indices rather than datetime indices

Reply:
For the purposes here, we just want to loop over the data, row by row. At this stage, I believe it isn't even important they are ordered chronologically.